src/test/java/seedu/recipe/logic/commands/RecipeDescriptorTest.java

```diff
@@ -1,36 +1,202 @@
-        // assertFalse(DESC_AMY.equals(editedAmy));
+    public void set_getDuration() {
+        assertDoesNotThrow(() -> testDescriptor.setDuration(TEST_DURATION));
+        assertEquals(Optional.of(TEST_DURATION), testDescriptor.getDuration());
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void set_getPortion() {
+        assertDoesNotThrow(() -> testDescriptor.setPortion(TEST_PORTION));
+        assertEquals(Optional.of(TEST_PORTION), testDescriptor.getPortion());
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void set_getTags() {
+        assertDoesNotThrow(() -> testDescriptor.setTags(TEST_TAGS));
+        assertEquals(Optional.of(TEST_TAGS), testDescriptor.getTags());
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void set_getIngredients() {
+        //Set from HashMap
+        assertDoesNotThrow(() -> testDescriptor.setIngredients(TEST_INGREDIENT_TABLE));
+        assertEquals(testDescriptor.getIngredients(), Optional.of(TEST_INGREDIENT_TABLE));
+
+        //Set from IngredientBuilder List
+        assertDoesNotThrow(() -> testDescriptor.setIngredients(TEST_INGREDIENT_LIST));
+        assertEquals(Optional.of(TEST_BUILDER.build()), testDescriptor.getIngredients());
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void set_getSteps() {
+        assertDoesNotThrow(() -> testDescriptor.setSteps(TEST_STEPS));
+        assertEquals(Optional.of(TEST_STEPS), testDescriptor.getSteps());
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void setPortionChanged() {
+        assertFalse(testDescriptor.isAnyFieldEdited());
+        testDescriptor.setPortionChanged(false);
+        assertFalse(testDescriptor.isAnyFieldEdited());
+        testDescriptor.setPortionChanged(true);
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void setDurationChanged() {
+        assertFalse(testDescriptor.isAnyFieldEdited());
+        testDescriptor.setDurationChanged(false);
+        assertFalse(testDescriptor.isAnyFieldEdited());
+        testDescriptor.setDurationChanged(true);
+        assertTrue(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void constructor_otherRecipeDescriptor_allFieldsCopied() {
+        //Set up test descriptor
+        testDescriptor.setName(TEST_NAME);
+        testDescriptor.setDuration(TEST_DURATION);
+        testDescriptor.setPortion(TEST_PORTION);
+        testDescriptor.setTags(TEST_TAGS);
+        testDescriptor.setIngredients(TEST_INGREDIENT_TABLE);
+        testDescriptor.setSteps(TEST_STEPS);
+
+        RecipeDescriptor actual = new RecipeDescriptor(testDescriptor);
+        assertEquals(testDescriptor, actual);
+    }
+
+    @Test
+    public void fieldsChanged_toRecipe_allFieldsPresent() {
```

Comment:
This method is more on confirming the fields have been copied correctly instead of present right. In that case the method name could be changed to smth like `fieldsChanged_toRecipe_allFieldsCopied`

Reply:
fixed